server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4320,7 +4320,14 @@ public boolean checkIfDynamicScalingCanBeEnabled(VirtualMachine vm, ServiceOffer
      */
     protected long configureCustomRootDiskSize(Map<String, String> customParameters, VMTemplateVO template, HypervisorType hypervisorType, DiskOfferingVO rootDiskOffering) {
         verifyIfHypervisorSupportsRootdiskSizeOverride(hypervisorType);
-        long rootDiskSizeInBytes = verifyAndGetDiskSize(rootDiskOffering, NumbersUtil.parseLong(customParameters.get(VmDetailConstants.ROOT_DISK_SIZE), -1));
+        Long rootDiskSizeCustomParam = null;
+        if (customParameters.containsKey(VmDetailConstants.ROOT_DISK_SIZE)) {
+            rootDiskSizeCustomParam = NumbersUtil.parseLong(customParameters.get(VmDetailConstants.ROOT_DISK_SIZE), -1);
+            if (rootDiskSizeCustomParam <= 0) {
+                throw new InvalidParameterValueException("Root disk size should be a positive number.");
```

Comment:
this is not the text I see in the description @weizhouapache . any last minute changes?

Reply:
@DaanHoogland 
this error is thrown if root disk size is passed , but invalid.

if root disk size is required but not passed, the error is thrown by code

https://github.com/apache/cloudstack/blob/3b05797344c15a9fbd6bee4ecd38281b677f24a2/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java#L4310-L4316